### PR TITLE
DWN-33428 : Release improvements

### PR DESCRIPTION
**Pull request by hsmith-gresham**

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -15,38 +22,95 @@ pipeline {
...
             }
 			steps {
-				sh "mvn versions:set -DnewVersion=${env.BRANCH_NAME}.GRESHAM-SNAPSHOT"
-				sh "mvn -N versions:update-child-modules"
 				timeout(time: 10, unit: 'MINUTES') {
-                	sh "mvn -B -V -U -T4 clean deploy -DaltSnapshotDeploymentRepository=snapshots::default::https://nexus.greshamtech.com/content/repositories/third-party-snapshots/"
+					withMaven(options: [junitPublisher(disabled: true)]) {
+						sh "mvn versions:set -B -DnewVersion=${env.BRANCH_NAME}.GRESHAM-SNAPSHOT"
+	                    sh "mvn -N -B versions:update-child-modules"
+	                    sh "mvn -B -V -U -T4 clean deploy -DaltSnapshotDeploymentRepository=snapshots::default::https://nexus.greshamtech.com/content/repositories/third-party-snapshots/"
```
Comment (asheppard-work) [inline Jenkinsfile]: `/content/repositories/third-party-snapshots/` is the old style URL.
It should be `/repository/thirdparty-maven-snapshots`

File: Jenkinsfile [hunk 2]
```diff
@@ -15,38 +22,95 @@ pipeline {
...
+					echo 'Current Version: ' + currentVersion
+
+					VERSION = currentVersion.substring(0, currentVersion.indexOf('-SNAPSHOT'))
+					echo 'Release Version: ' + VERSION
+
+					def parts = VERSION.tokenize('-')
+					echo 'Parts: ' + parts
+
+					def index = parts[1].toInteger()
+					echo 'Index: ' + index
```
Comment (asheppard-work) [inline Jenkinsfile]: What is index? Is it actually the release/version number?

File: Jenkinsfile [hunk 3]
```diff
@@ -58,6 +122,20 @@ pipeline {
...
+		stage ('Bump Development Version') {
+			when {
+				expression {
+					return BRANCH_NAME == '1.3.x' && params.RELEASE
+				}
+			}
+			steps {
+				script {
+					sh "mvn versions:set -DnewVersion=${NEW_VERSION}-SNAPSHOT --batch-mode"
+                    sh "git commit --all --message 'New Development Version $NEW_VERSION-SNAPSHOT'"
```
Comment (asheppard-work) [inline Jenkinsfile]: Indentation does go a bit wonky in this file.

**Review by asheppard-work — CHANGES_REQUESTED** [review]

File: Jenkinsfile [hunk 4]
```diff
@@ -1,12 +1,19 @@
 //
-// CTC Auth Jenkins Pipeline
+// CTC Open Id Connect Jenkins Pipeline
```
Comment (asheppard-work) [inline Jenkinsfile]: I don't know why we bother with this comment, the location of the file tells you what it is! I removed them from the cashproof repo.

This is also a classic case of a comment getting out of sync. It doesn't tell us anything useful, remove!

File: Jenkinsfile (hunk 1 above)
Comment (hsmith-gresham) [reply Jenkinsfile]: I thought it looked odd, will change

File: Jenkinsfile (hunk 2 above)
Comment (hsmith-gresham) [reply Jenkinsfile]: Yeah, will rename, will also remove the echos as they were just there for testing purposes

File: Jenkinsfile (hunk 3 above)
Comment (hsmith-gresham) [reply Jenkinsfile]: Indeed it does - will do a separate whitespace commit

File: Jenkinsfile (hunk 4 above)
Comment (hsmith-gresham) [reply Jenkinsfile]: Will remove

File: Jenkinsfile [hunk 5]
```diff
@@ -15,38 +22,95 @@ pipeline {
...
+				}
+                timeout(time: 10, unit: 'MINUTES') {
+                    withMaven(options: [jUnitPublisher(disabled: true)]) {
+	                    sh "mvn -B -V -U -T4 clean deploy -DaltReleaseDeploymentRepository=releases::default::https://nexus.greshamtech.com/content/repositories/thirdparty/"
+                    }
+                }
+            }
+            post {
+                always {
+                       archiveArtifacts caseSensitive: false, onlyIfSuccessful: true, allowEmptyArchive: true, artifacts: 'openid-connect-server-webapp/target/*.war'
```
Comment (asheppard-work) [inline Jenkinsfile]: Given you have wrapped the mvn deploy step with `withMaven`, you shouldn't need to manually archive artifacts. Jenkins will make a magic link to Nexus.

Comment (hsmith-gresham) [reply Jenkinsfile]: Just mimicked from what was existing in the Jenkinsfile but get your point, will remove

**Review by asheppard-work — APPROVED** [review]

**Review by gresham-Carling — APPROVED** [review]

**Review by gkatiyar — APPROVED** [review]

**Pull request merged**